src/Messaging/Events/TaskCallbackEvent.cs

```diff
@@ -49,14 +49,21 @@ public class TaskCallbackEvent : EventBase
         [JsonProperty(PropertyName = "metadata")]
         public Dictionary<string, object> Metadata { get; set; }
 
-        public TaskCallbackEvent()
+        /// <summary>
+        /// Gets or sets any output artifacts generated by the task.
+        /// </summary>
+        [JsonProperty(PropertyName = "output_artifacts")]
+        public Dictionary<string, string> OutputArtifacts { get; set; }
+
+    public TaskCallbackEvent()
         {
             WorkflowInstanceId = String.Empty;
             TaskId = String.Empty;
             ExecutionId = String.Empty;
             CorrelationId = String.Empty;
             Identity = String.Empty;
             Metadata = new Dictionary<string, object>();
+            OutputArtifacts = new Dictionary<string, string>();
```

Comment:
@jackschofield23 I think we should match the schema of the task dispatch event, and use storage objects rather than a simple dictionary